src/main/java/mil/dds/anet/resources/AdminResource.java

```diff
@@ -57,4 +71,107 @@ public Integer saveAdminSettings(@GraphQLRootContext Map<String, Object> context
     return config.getDictionary();
   }
 
+  /**
+   * If anet-dictionary.yml file is changed manually while ANET is up and running ,this method can
+   * be used to reload the dictionary with new values without restarting the server
+   */
+  @GraphQLQuery(name = "reloadDictionary")
+  public String reloadDictionary(@GraphQLRootContext Map<String, Object> context) {
+    final Person user = DaoUtils.getUserFromContext(context);
+    AuthUtils.assertAdministrator(user);
+    config.reloadDictionary();
+    AnetAuditLogger.log("Dictionary updated by {}", user);
+    return AnetConstants.DICTIONARY_RELOAD_MESSAGE;
+  }
+
+  /**
+   * Clears Domain Users Cache
+   */
+  @GraphQLQuery(name = "clearCache")
+  public String clearCache(@GraphQLRootContext Map<String, Object> context) {
+    final Person user = DaoUtils.getUserFromContext(context);
+    AuthUtils.assertAdministrator(user);
+    return AnetObjectEngine.getInstance().getPersonDao().clearCache();
+  }
+
+  /**
+   * Returns the project version which is saved during project build (See project.version definition
+   * in build.gradle file) Right after project information is written into version.properties file
+   * on startup,it is read and set with AnetConfiguration loadVersion method
+   */
+  @GET
+  @Timed
+  @Path("/projectVersion")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Map<String, Object> getProjectVersion() {
+    return new HashMap() {
+      {
+        put("projectVersion", config.getVersion());
+      }
+    };
+  }
+
+  /**
+   * Returns user logs in descending order of time
+   */
+  @GraphQLQuery(name = "userActivities")
+  public Map<String, Object> userActivities(@GraphQLRootContext Map<String, Object> context)
```

Comment:
we should not parse log files to implement services.
we could store the user activity in the java bean of people as they are cached

Reply:
Now user activities are stored in cached Person bean rather than a log file